src/rust/receiver-mock/src/main.rs

```diff
@@ -1,259 +1,17 @@
-                  }
-                }
-
-                let metrics_ip_list = (*stats).metrics_ip_list.entry(address).or_insert(0);
-                *metrics_ip_list += 1;
-              }
-            },
-            // Logs & events
-            "application/x-www-form-urlencoded" => {
-              let mut stats = statistics.lock().unwrap();
-
-              if (*stats).print_logs {
-                let mut counter = 0;
-                for line in lines {
-                  println!("log => {}", line);
-                  counter += 1;
-                }
-                (*stats).logs += counter;
-                (*stats).logs_bytes += vector_length;
-
-                let logs_ip_list = (*stats).logs_ip_list.entry(address).or_insert((0, 0));
-                (*logs_ip_list).0 += counter;
-                (*logs_ip_list).1 += vector_length;
-              }
-              else {
-                let lines_count = lines.count() as u64;
-                (*stats).logs_bytes += vector_length;
-                (*stats).logs += lines_count;
-
-                let logs_ip_list = (*stats).logs_ip_list.entry(address).or_insert((0, 0));
-                (*logs_ip_list).0 += lines_count;
-                (*logs_ip_list).1 += vector_length;
-              }
-
-            },
-            &_ => {
-              println!("invalid header value");
-            }
-          }
-          stats(statistics);
-          Ok(Response::new("".into()))
-        }
-      }
-    }
-}
-
-async fn run_app(statistics: Arc<Mutex<Statistics>>, port: u16) {
-    let addr = SocketAddr::from(([0, 0, 0, 0], port));
-    println!("Receiver mock is waiting for enemy on 0.0.0.0:{}!", port);
-    let make_svc = make_service_fn(|conn: &AddrStream | {
-      let statistics = statistics.clone();
-      let address = conn.remote_addr().ip();
-      async move {
-        let statistics = statistics.clone();
-        let result = service_fn(move |req| handle(
-          req,
-          address,
-          statistics.clone()
-        ));
-        Ok::<_, Infallible>(result)
-    }});
 
-    let server = Server::bind(&addr).serve(make_svc);
-
-    if let Err(e) = server.await {
-        eprintln!("server error: {}", e);
-    }
-
-}
-
-fn stats(statistics: Arc<Mutex<Statistics>>) {
-  let mut statistics = statistics.lock().unwrap();
-
-  if get_now() >= (*statistics).ts + 60 {
-      println!("{} Metrics: {:10.} Logs: {:10.}; {:6.6} MB/s",
-        (*statistics).ts,
-        (*statistics).metrics - (*statistics).p_metrics,
-        (*statistics).logs - (*statistics).p_logs,
-        (((*statistics).logs_bytes - (*statistics).p_logs_bytes) as f64)/((get_now()-(*statistics).ts) as f64)/(1e6 as f64));
-      (*statistics).ts = get_now();
-      (*statistics).p_metrics = (*statistics).metrics;
-      (*statistics).p_logs = (*statistics).logs;
-      (*statistics).p_logs_bytes = (*statistics).logs_bytes;
-  }
-}
+mod metrics;
+mod router;
+mod statistics;
+use statistics::Statistics;
```

Comment:
I'd like to know:

* why do I need both of those here? When I skip `mod statistics` I get
  ```
  unresolved import `statistics`

  use of undeclared type or module `statistics`rustc(E0432)
  ```
* I can use `use crate::statistics;` in e.g. `router.rs` but in `main.rs` when I use it I get
  ```
  unresolved import `crate::statistics`

  no `statistics` in the rootrustc(E0432)
  ```

@sumo-drosiek any ideas?

Reply:
I put both `mod` and `use` and cargo returns this:

```
error[E0255]: the name `statistics` is defined multiple times
  --> src/main.rs:15:5
   |
12 | mod statistics;
   | --------------- previous definition of the module `statistics` here
...
15 | use crate::statistics;
   |     ^^^^^^^^^^^^^^^^^ `statistics` reimported here
   |
   = note: `statistics` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
15 | use crate::statistics as other_statistics;
```

You need to declare `mod statistics` somewhere, so you can use `crate::statistics` in other files, and you cannot overwrite module by importing another one in it's place